File: old/src/ops.ts
export enum UnaryOps { EXP2 = "EXP2", LOG2 = "LOG2", CAST = "CAST", BITCAST = "BITCAST", SIN = "SIN", SQRT = "SQRT", RECIP = "RECIP", NEG = "NEG" };
export enum BinaryOps { ADD = "ADD", MUL = "MUL", IDIV = "IDIV", MAX = "MAX", MOD = "MOD", CMPLT = "CMPLT", CMPNE = "CMPNE", XOR = "XOR", SHL = "SHL", SHR = "SHR", OR = "OR", AND = "AND", THREEFRY = "THREEFRY", SUB = "SUB" };
export enum TernaryOps { WHERE = "WHERE", MULACC = "MULACC" };
export enum ReduceOps { SUM = "SUM", PROD = "PROD", MAX = "MAX" };
export enum MetaOps { EMPTY = "EMPTY", CONST = "CONST", COPY = "COPY", CONTIGUOUS = "CONTIGUOUS", ASSIGN = "ASSIGN", VIEW = "VIEW" };
export type Ops = UnaryOps | BinaryOps | TernaryOps | ReduceOps | MetaOps;

File: ops.py
from enum import IntEnum


class UnaryOps(IntEnum):
    EXP2 = 100; LOG2 = 101; CAST = 102; BITCAST = 103; SIN = 104; SQRT = 105; RECIP = 106; NEG = 107

class BinaryOps(IntEnum):
    ADD = 200; MUL = 201; IDIV = 202; MAX = 203; MOD = 204; CMPLT = 205; CMPNE = 206; XOR = 207
    SHL = 208; SHR = 209; OR = 210; AND = 211; THREEFRY = 212; SUB = 213

class TernaryOps(IntEnum):
    WHERE = 300; MULACC = 301

class ReduceOps(IntEnum):
    SUM = 400; PROD = 401; MAX = 402

class MetaOps(IntEnum):
    EMPTY = 500; CONST = 501; COPY = 502; CONTIGUOUS = 503; ASSIGN = 504; VIEW = 505

Op = (*UnaryOps, *BinaryOps, *TernaryOps, *ReduceOps, *MetaOps)